Source/CMakeLists.txt

```diff
@@ -19,10 +19,11 @@ endif()
 add_definitions(-D__STDC_LIMIT_MACROS)
 add_definitions(-D__STDC_CONSTANT_MACROS)
 
-macro(add_dolphin_library lib srcs libs)
+function(add_dolphin_library lib srcs libs)
 	add_library(${lib} STATIC ${srcs})
 	target_link_libraries(${lib} PUBLIC ${libs})
-endmacro()
+	set_target_source_folder(${lib})
```

Comment:
When I say "breaks," I mean that it doesn't correctly add all sources to the right folders in the IDE. It still compiles fine.

In AudioCommon, `OpenALStream.cpp`, `aldlist.cpp`, and `XAudioStream.cpp` are all grouped in the CMake-default "Source Files" folder instead of at project toplevel like the rest of the sources.